packages/dashboard/src/dashboard-actions/dashboardReducer.ts

```diff
@@ -170,7 +171,7 @@ export const dashboardReducer: Reducer<DashboardState, DashboardAction> = (
 
     case 'UPDATE':
       return {
-        ...(state = { ...state, ...action.payload.fieldsToUpdate }),
+        ...(state = update({ state, fieldsToUpdate: action.payload.fieldsToUpdate })),
```

Comment:
We added a lot of code and tests just to perform a deep merge. Why can't we call deep merge here and add a single test for this action? Way less to maintain and I don't see a value add by doing it this way instead.